### DISCUSSION TITLE
Calling `x-data` on `<html>` returns an error

### DISCUSSION THREAD

**Discussion by jordanthornquest** [alpinejs/alpine#1457] [7. Bugs (archived)]
Hello folks,

I'm looking to toggle a class on the `<html>` tag with Alpine. However, it appears initializing `x-data` like this:

```html
<html x-data>
  <body>
    <script defer src="https://cdn.jsdelivr.net/npm/alpinejs@3.0.1/dist/cdn.min.js"></script>
  </body>
</html>
```

Gives me this error in the browser console:

```
Uncaught TypeError: e.matches is not a function
    D https://cdn.jsdelivr.net/npm/alpinejs@3.0.1/dist/cdn.min.js:5
    D https://cdn.jsdelivr.net/npm/alpinejs@3.0.1/dist/cdn.min.js:5
    e https://cdn.jsdelivr.net/npm/alpinejs@3.0.1/dist/cdn.min.js:5
    Lt https://cdn.jsdelivr.net/npm/alpinejs@3.0.1/dist/cdn.min.js:5
    <anonymous> https://cdn.jsdelivr.net/npm/alpinejs@3.0.1/dist/cdn.min.js:5
    VoidFunction* https://cdn.jsdelivr.net/npm/alpinejs@3.0.1/dist/cdn.min.js:5
    <anonymous> https://cdn.jsdelivr.net/npm/alpinejs@3.0.1/dist/cdn.min.js:5
```

I'm using Firefox on MacOS Big Sur 11.4. This is with AlpineJS `v3.0.1`. Anything I'm missing here?

**Comment by KevinBatdorf** [comment]
Looking at it quickly, it looks like Alpine is looking for a parentElement, which the `html` element doesn't have. 

https://github.com/alpinejs/alpine/blob/b46c41fa240cd8af2dcaa29fb60fb1db0389c95a/packages/alpinejs/src/lifecycle.js#L41

Could probably switch it to parentNode or maybe just return early by moving line 44 up a bit, but I'm not sure. 

If you want some help on a workaround, share the use case or a demo and I can probably suggest an alternative approach.

**Reply by jordanthornquest** [reply]
Hey, thanks for looking into this. I'm wanting to add a `dark` class to the `<html>` element for a good ol' dark mode toggle.

**Reply by SimoTod** [reply]
You can add it to the body for now
